pkg/pillar/types/volumetypes.go

```diff
@@ -286,20 +285,29 @@ func (config VolumesSnapshotConfig) Key() string {
 
 // VolumesSnapshotStatus is used to send snapshot status from volumemgr to zedmanager
 type VolumesSnapshotStatus struct {
-	// SnapshotID is the ID of the snapshot
+	// SnapshotID is the ID of the snapshot, critical field
 	SnapshotID string
-	// Metadata is a map of volumeID to metadata, depending on the volume type
+	// Metadata is a map of volumeID to metadata, depending on the volume type. Critical field.
 	VolumeSnapshotMeta map[string]interface{}
 	// TimeCreated is the time the snapshot was created, reported by FS-specific code
 	TimeCreated time.Time
-	// AppUUID used as a backlink to the app
+	// AppUUID used as a backlink to the app, critical field
 	AppUUID uuid.UUID
 	// RefCount is the number of times the snapshot is used. Necessary to trigger the handleModify handler
 	RefCount int
+	// ResultOfAction is the type of action that was performed on the snapshot that resulted in this status
+	ResultOfAction VolumesSnapshotAction
 	// ErrorAndTimeWithSource provides SetErrorNow() and ClearError()
 	ErrorAndTimeWithSource
 }
 
+// VolumesSnapshotStatusCriticalFields are the fields that must present in a serialized snapshot status
```

Comment:
Since the json Encode omits fields which have their default values it might make sense to add to this comment that such fields must have a non-default value?

Reply:
Sorry my confusion with protobuf encoding.